src/lib/containers/entity/metadata/EntityModal.tsx

```diff
@@ -62,36 +66,90 @@ export const EntityModal: React.FC<EntityModalProps> = ({
     isVersionable && (entityBundle.entity as VersionableEntity).isLatestVersion!
 
   useEffect(() => {
-    if (show) {
-      rebuildTooltip()
+    rebuildTooltip()
+  })
+
+  let primaryAction
+  let tertiaryActions
+
+  if (!entityBundle) {
+    primaryAction = { skeleton: true }
+    tertiaryActions = undefined
+  } else {
+    // TODO: Determine if we're on the entity page to conditonally show this button
+    primaryAction = {
+      skeleton: false,
+      copy: `Open ${entityTypeToFriendlyName(entityBundle.entityType!)}`,
+      onClick: () =>
+        window.open(
+          `${getEndpoint(
+            BackendDestinationEnum.PORTAL_ENDPOINT,
+          )}#!Synapse:${entityId}`,
+          '_blank',
+          'noopener',
+        ),
     }
-  }, [show])
+  }
+
+  if (currentTab === EntityModalTabs.ANNOTATIONS) {
+    if (isInEditMode) {
+      primaryAction = {
+        copy: `Save Annotations`,
+        onClick: () => {
+          annotationEditorFormRef.current?.submit()
+        },
+      }
+      tertiaryActions = [
+        {
+          copy: 'Cancel',
+          onClick: () => {
+            setIsInEditMode(false)
+          },
+        },
+      ]
+    } else if (canEdit) {
+      tertiaryActions = [
+        {
+          copy: 'Edit',
+          disabled: isVersionable && !isLatestVersion,
+          'data-for': 'entityModalTooltip',
+          'data-tip':
+            isVersionable && !isLatestVersion
+              ? 'Annotations can only be edited on the latest version'
+              : undefined,
+          onClick: () => {
+            setIsInEditMode(true)
+          },
+        },
+      ]
+    }
+  }
```

Comment:
As this modal becomes more complex, this behavior should get pulled into helper functions. For now, I feel that that refactor would be more complex than this existing code, so I didn't do it yet.